components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/backchannellogout/LogoutRequestSender.java

```diff
@@ -52,12 +57,56 @@
 public class LogoutRequestSender {
 
     private static final Log LOG = LogFactory.getLog(LogoutRequestSender.class);
-    private static ExecutorService threadPool = Executors.newFixedThreadPool(2);
+    private static ExecutorService threadPool = null;
+    private boolean hostNameVerificationEnabled = true;
+    private static int httpConnectTimeout = 0;
+    private static int httpSocketTimeout = 0;
+
     private static LogoutRequestSender instance = new LogoutRequestSender();
     private static final String LOGOUT_TOKEN = "logout_token";
 
     private LogoutRequestSender() {
 
+        String poolSize = IdentityUtil.getProperty(OIDCSessionConstants.OIDCLogoutRequestConstants.POOL_SIZE);
+        String keepAliveTime = IdentityUtil.getProperty(
+                OIDCSessionConstants.OIDCLogoutRequestConstants.KEEP_ALIVE_TIME);
+        String httpConnectTimeoutProperty = IdentityUtil.getProperty(
+                OIDCSessionConstants.OIDCLogoutRequestConstants.HTTP_CONNECT_TIMEOUT);
+        String httpSocketTimeoutProperty = IdentityUtil.getProperty(
+                OIDCSessionConstants.OIDCLogoutRequestConstants.HTTP_SOCKET_TIMEOUT);
+        String hostNameVerificationEnabledProperty = IdentityUtil.getProperty(
+                IdentityConstants.ServerConfig.SLO_HOST_NAME_VERIFICATION_ENABLED);
+
+        if (poolSize == null) {
+            poolSize = OIDCSessionConstants.OIDCLogoutRequestConstants.DEFAULT_POOL_SIZE;
+        }
+        if (keepAliveTime == null) {
+            keepAliveTime = OIDCSessionConstants.OIDCLogoutRequestConstants.DEFAULT_KEEP_ALIVE_TIME;
+        }
+        if (httpConnectTimeoutProperty == null) {
+            httpConnectTimeoutProperty = OIDCSessionConstants.OIDCLogoutRequestConstants.DEFAULT_HTTP_CONNECT_TIMEOUT;
+        }
+        if (httpSocketTimeoutProperty == null) {
+            httpSocketTimeoutProperty = OIDCSessionConstants.OIDCLogoutRequestConstants.DEFAULT_HTTP_SOCKET_TIMEOUT;
+        }
+
+        int poolSizeInt = Integer.parseInt(poolSize);
+        int keepAliveTimeLong = Integer.parseInt(keepAliveTime);
+        threadPool = new ThreadPoolExecutor(poolSizeInt, poolSizeInt, keepAliveTimeLong,
```

Comment:
Don't we need to define the queue size as well? Otherwise this can lead to OOM in some rare scenarios.

Reply:
Addressed via 696835c1e65a4a4e34981c2dfe9bc7c9ccd23e71